Restrict nickname size and characters

RitzKid76 opened the pull request:
This now blocks:
- 17+ char length (excluding color codes)
- chars that aren't usually allowed by Minecraft name standards

RitzKid76 committed:
restrict nicknames size and character possiblilities

this now blocks:
- 17+ char length (excluding color codes)
- chars that arent normally allowed by minecraft name standards

RitzKid76 committed:
clean up Commandnick file

Grassboii reviewed (changes requested):
You should allow for one pair of '[]', to ease into the transition of not allowing special characters
on top of that, you should also make the limit 20 characters, so 21 isn't allowed

RitzKid76 committed:
implement the requested changes

allows at least 1 of each '[' and ']'
it was discussed on voice chat that 20 characters was too much, and 16 was more than fine

RitzKid76 committed:
fix a minor bug

RitzKid76 committed:
added new feedback messages for setting others nicknames

sshshark reviewed (approved):
Works as intended.
